src/global/reco/ReconstructedElectrons_factory.h

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2022, 2023 Daniel Brandenburg
+// Subject to the terms in the LICENSE file found in the top-level directory.
+
+#pragma once
+
+// JANA
+#include <extensions/jana/JChainFactoryT.h>
+#include <JANA/JEvent.h>
+
+// data model
+#include <edm4eic/TrackSegmentCollection.h>
+
+// algorithms
+#include <algorithms/reco/ElectronReconstruction.h>
+
+// services
+#include <services/log/Log_service.h>
+#include <extensions/spdlog/SpdlogExtensions.h>
+#include <extensions/spdlog/SpdlogMixin.h>
+
+namespace eicrecon {
+
+// algorithm
+  class ElectronReconstruction;
+
```

Comment:
Forward declaration not needed. You embed the algorithm as non-pointer member, so header inclusion is required.
```suggestion
```